src/main/java/appeng/parts/p2p/PartP2PFluids.java

```diff
@@ -85,11 +89,54 @@ public void onNeighborChanged()
 	}
 
 	@Override
-	public int fill( final EnumFacing from, final FluidStack resource, final boolean doFill )
+	public boolean hasCapability( Capability<?> capabilityClass )
+	{
+		if( capabilityClass == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY )
+		{
+			return true;
+		}
+
+		return super.hasCapability( capabilityClass );
+	}
+
+	@Override
+	public <T> T getCapability( Capability<T> capabilityClass )
+	{
+		if( capabilityClass == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY )
+		{
+			return (T) this;
+		}
+
+		return super.getCapability( capabilityClass );
+	}
+
+	@Override
+	public IPartModel getStaticModels()
+	{
+		return MODELS.getModel( isPowered(), isActive() );
+	}
+
+	@Override
+	public IFluidTankProperties[] getTankProperties()
+	{
+		if( !this.isOutput() )
+		{
+			final PartP2PFluids tun = this.getInput();
+			if( tun != null )
+			{
+				return ACTIVE_TANK;
+			}
+		}
+
+		return INACTIVE_TANK;
+	}
+
+	@Override
+	public int fill( FluidStack resource, boolean doFill )
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "fill" is 20 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)